src/storage.rs

```diff
@@ -68,16 +71,34 @@ impl Default for AccountBasic {
     }
 }
 
-/// EVM Code, currently mapping to REVM's [ByteCode::LegacyAnalyzed].
-// TODO: Support raw legacy & EOF
+/// Analyzed legacy code.
 #[derive(Debug, Clone, PartialEq, Default, Serialize, Deserialize)]
-pub struct EvmCode {
-    /// Bytecode with 32 zero bytes padding
-    bytecode: Bytes,
-    /// Original bytes length
-    original_len: usize,
+pub struct LegacyCode {
+    /// Bytecode with 32 zero bytes padding.
+    pub bytecode: Bytes,
```

Comment:
Let's not make these fields `pub` as that allows people to construct an invalid `LegacyCode` in memory. For now, let's remove the `Default`s and guard the only entry of `revm` conversion well. In the future, we can add `new` functions that do proper validation and analysis and return a `Result`.

Reply:
Good callout.